src/plugins/kibana_usage_collection/server/collectors/cloud/detector/gcp.ts

```diff
@@ -21,61 +20,57 @@ const SERVICE_ENDPOINT = 'http://169.254.169.254/computeMetadata/v1/instance';
  * @internal
  */
 export class GCPCloudService extends CloudService {
-  constructor(options: CloudServiceOptions = {}) {
-    super('gcp', options);
+  constructor() {
+    super('gcp');
   }
 
-  _checkIfService(request: Request) {
+  protected _checkIfService = async () => {
     // we need to call GCP individually for each field we want metadata for
     const fields = ['id', 'machine-type', 'zone'];
 
-    const create = this._createRequestForField;
-    const allRequests = fields.map((field) => promisify(request)(create(field)));
-    return (
-      Promise.all(allRequests)
-        // Note: there is no fallback option for GCP;
-        // responses are arrays containing [fullResponse, body];
-        // because GCP returns plaintext, we have no way of validating
-        // without using the response code.
-        .then((responses) => {
-          return responses.map((response) => {
-            if (!response || response.statusCode === 404) {
-              throw new Error('GCP request failed');
-            }
-            return this._extractBody(response, response.body);
-          });
-        })
-        .then(([id, machineType, zone]) => this._combineResponses(id, machineType, zone))
+    const settledResponses = await Promise.allSettled(
+      fields.map(async (field) => {
+        return await fetch(`${SERVICE_ENDPOINT}/${field}`, {
+          method: 'GET',
+          headers: {
+            // GCP requires this header
+            'Metadata-Flavor': 'Google',
```

Comment:
Could we please extract the GCP required header into a const, similar to what we did for the `SERVICE_ENDPOINT`? That way, it's easier to search for, tightens up the constants we're declaring and less likely to get lost as changes are made with time.

e.g. something along the lines of 
```
const SERVICE_ENDPOINT = 'http://169.254.169.254/computeMetadata/v1/instance';
const SERVICE_HEADER = 'metadata-Flavor';

const getServiceHeaders = () => ({ SERVICE_HEADER: 'Google' });
```
with 
```
return await fetch(`${SERVICE_ENDPOINT}/${field}`, {
  method: 'GET',
  headers: {
    // GCP requires this header
    ...getServiceHeaders(),
  },
});
```
I'm sure you can come up with an improvement here 😄

Reply:
Makes sense :+1: